### PR TITLE
AMP implementation of Media.net header bidder

### PR DESCRIPTION
**Pull request by adityap2016**
Please find the Media.net implementation for amp-ad

We have a Google Corporate CLA. 
Organization Name - Media.net Advertising FZ-LLC

### CODE REVIEW

**Comment by adityap2016** [comment]
I signed it!

**Comment by lannka** [comment]
@adityap2016 
- could you implement the render-start and no-content APIs? see [developer guidelines](https://github.com/ampproject/amphtml/blob/master/ads/README.md#developer-guidelines-for-a-pull-request).
- seems you didn't get through the CLA robot check. could you follow the instruction to see if you can get it resolved?

**Comment by adityap2016** [comment]
@lannka  

- We plan to implement the render-start and no-content APIs in our next release. 

- The Corporate CLA was signed by Vaibhav Arya. Organisation Name - Media.net Advertising FZ-LLC

**Comment by adityap2016** [comment]
@lannka  - Looks like our CLA got approved. Let us know if there is anything else that you would require.

**Comment by lannka** [comment]
@adityap2016 does media.net always use doubleclick? if so, you can add `renderStartImplemented=true` to the `_config.js`.

And very occasionally (1 in 10 times), i see blank ad. Do you know what could be a cause?
Tip refresh this in browser: `http://localhost:8000/examples/ads.amp.max.html?type=medianet`

**Comment by monish-chhadwa** [comment]
I confirm that I am okay with these commits being contributed to Google!

**Comment by adityap2016** [comment]
@lannka  - We have added renderstartimplemented = true in the config. Additionally, we investigated the the blank ad issue and the fix required customising the DFP line item setup. This should not happen anymore. Please let me know in case of further queries.

**Comment by lannka** [comment]
@adityap2016 seems to be bad rebase, 152 files are touched. Can you try again?

**Comment by ruturajv-media-net** [comment]
I signed it!

**Comment by adityap2016** [comment]
@lannka - We have fixed the bad rebase. Please take a look.

File: ads/medianet.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      typeof global.advBidxc.registerAmpSlot === 'function') {
+      global.advBidxc.registerAmpSlot({
+        cb: loadDFP,
+        data,
+        winObj: global,
+      });
+    }
+  }
+
+  global.setTimeout(function() {
```
Comment (lannka) [inline ads/medianet.js]: use arrow function `() => {}`

File: ads/medianet.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    loadDFP();
+  }, data.timeout || dfpDefaultTimeout);
+
+  computeInMasterFrame(global, 'medianet-hb-load', function(done) {
```
Comment (lannka) [inline ads/medianet.js]: same here

File: ads/medianet.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+  function loadDFP() {
+    if (isDoubleClickCalled) {
+      return;
+    }
+    isDoubleClickCalled = true;
+
+    global.advBidxc = global.context.master.advBidxc;
+    if (global.advBidxc && typeof global.advBidxc.renderAmpAd === 'function') {
+      global.addEventListener('message', function(event) {
```
Comment (lannka) [inline ads/medianet.js]: use arrow function `event => {...}`

**Comment by adityap2016** [comment]
@lannka  - we have made the necessary changes.

**Comment by adityap2016** [comment]
@lannka  - We have removed ampslotindex. Please take a look.

**Pull request merged**